Refactored `rotation_mode` in Node3D

TokageItLab opened the pull request:
Fixed #62225. Contained fix for non-orthonormalized Quaternion bug.

`set_euler_scale()` in `Node3D::_update_local_transform()` has not work and reset rotation value in Quaternion and Basis mode in Node3D. This PR fixed it.

reduz commented on scene/3d/node_3d.cpp:
```diff
@@ -88,7 +88,9 @@ void Node3D::_update_local_transform() const {
 	}
-	data.local_transform.basis.set_euler_scale(data.rotation, data.scale);
+	if (this->get_rotation_edit_mode() == ROTATION_EDIT_MODE_EULER) {
+		data.local_transform.basis.set_euler_scale(data.rotation, data.scale);
```
The whole code of this function looks wrong for some reason. Why setting local_transform if its going to be overriden later? Also setting the euler like this seems to ignore the rotation order of the node.

TokageItLab replied:
Indeed. It seems that DIRTY_LOCAL is not needed in the first place. I will attempt to refactor it.

TokageItLab commented:
@reduz Refactored. `DIRTY_LOCAL` has gone. And the `local_transform` may temporarily hold a non-orthogonal matrix, but it will be orthogonalized by `get_transform()` as the getter.

reduz commented:
@TokageItLab No, thats not right either. Those variables do need to exist because it allows you to set rotation and scale separately without losing floating point precision. The problem is in that function alone from what I can gather.

TokageItLab committed:
fix method when rotation_mode changed

TokageItLab commented:
> Those variables do need to exist because it allows you to set rotation and scale separately without losing floating point precision.

@reduz You have a point. Finally, I make changes as follow:

- Fixed `_update_local_transform()`
- In `set_rotation_edit_mode()`, include conversion when changing from Basis to other modes to prevent scale and rotation being reset; if there is never a change to Basis mode, there should be no effect on those values
- Replaced `data.local_transform` with `get_transform()` to prevent `_update_local_transform()` validation from missing

TokageItLab commented:
@reduz Ready for review. For consistency, I added a quaternion separate from local_transform as well as rotation, scale. Without it, it is not possible to separate quaternion from scale[^1].

[^1]: It is not possible to both set scale to quaternion when loading a scene and to operate scale independently of quaternion.

TokageItLab commented on scene/3d/node_3d.cpp:
```diff
@@ -85,10 +85,14 @@ void Node3D::_notify_dirty() {
-	if (this->get_rotation_edit_mode() != ROTATION_EDIT_MODE_BASIS) {
-		data.local_transform = data.local_transform.orthogonalized();
+	if (this->get_rotation_edit_mode() == ROTATION_EDIT_MODE_EULER) {
+		data.local_transform.orthonormalize();
```
Indeed, set_euler() seems to orthonormalize Basis internally.

reduz commented:
I am very sorry, I misunderstood the underlying problem and gave bad feedback to Tokage, then realized everything was on the wrong path. This is my responsibility as I now realize I did a bad job with the initial PR implementing this feature, so I will take some time to understand how to fix this properly and do another pull request.
